Cross Site GossipRouter TLS: java.io.EOFException: SSL peer shut down incorrectly

Hi colleagues, 
I am using `infinispan-operator v.2.2.x` and trying to set up TLS for x-site GossipRouter following [your documentation](https://infinispan.org/docs/infinispan-operator/2.2.x/operator.html#securing-cross-site-connections_cross-site)
, and getting an error ` java.io.EOFException: SSL peer shut down incorrectly` inside the `mycluster-router` POD

Could you explain what is wrong with my configuration (probably with my keystore inside the [x-site secret](https://infinispan.org/docs/infinispan-operator/2.2.x/operator.html#cross-site-encryption-secrets_cross-site)) ?

Configuration details below:



**mycluster(env-vpronkin) site**

```sh
kubectl get secret -n infinispan mycluster-tls-secret -o jsonpath='{.data.keystore\.p12}' | base64 --decode > keystore.p12
keytool -list -v -storetype pkcs12 -keystore keystore.p12 -storepass password

Keystore type: PKCS12
Keystore provider: SUN

Your keystore contains 1 entry

Alias name: mycluster
Creation date: Jul 5, 2022
Entry type: PrivateKeyEntry
Certificate chain length: 2
Certificate[1]:
Owner:
Issuer: CN=Base - cert-manager CA, O=Digital, ST=Some-State, C=AU
Serial number: 5302b77eeaedc5506fc71cca43a00ece
Valid from: Tue Jul 05 16:49:59 EEST 2022 until: Mon Oct 03 16:49:59 EEST 2022
Certificate fingerprints:
         SHA1: 1B:67:15:D6:E8:FE:A4:B1:6F:17:1B:93:B8:23:D9:94:08:FF:AA:9B
         SHA256: D1:9D:40:0F:54:CC:64:33:55:54:CD:B5:70:6C:90:BC:DA:F0:4D:07:BD:4F:7B:BE:96:67:4C:54:A4:D3:78:07
Signature algorithm name: SHA256withRSA
Subject Public Key Algorithm: 2048-bit RSA key
Version: 3

Extensions:

#1: ObjectId: 2.5.29.19 Criticality=true
BasicConstraints:[
  CA:false
  PathLen: undefined
]

#2: ObjectId: 2.5.29.15 Criticality=true
KeyUsage [
  DigitalSignature
  Key_Encipherment
]

#3: ObjectId: 2.5.29.17 Criticality=true
SubjectAlternativeName [
  DNSName: mycluster.infinispan.svc
  DNSName: mycluster.infinispan.svc.cluster.local
]

Certificate[2]:
Owner: CN=Base - cert-manager CA, O=Digital, ST=Some-State, C=AU
Issuer: CN=Base - Root CA, O=Digital, ST=Some-State, C=AU
Serial number: 3eac62fdbe71c450323654458d6fce4dec1fc64f
Valid from: Wed Apr 21 09:54:00 EEST 2021 until: Sat Apr 19 09:54:00 EEST 2031
Certificate fingerprints:
         SHA1: C7:95:CF:58:72:4D:AB:3A:E4:12:F6:3C:21:5B:2F:DE:C5:CB:FF:23
         SHA256: 96:3F:EA:78:E8:C5:86:8E:31:7B:54:6A:7A:52:0C:F3:33:13:73:24:D1:18:99:F8:15:24:70:69:B4:ED:EF:2E
Signature algorithm name: SHA256withRSA
Subject Public Key Algorithm: 2048-bit RSA key
Version: 3

Extensions:

#1: ObjectId: 2.5.29.35 Criticality=false
AuthorityKeyIdentifier [
KeyIdentifier [
0000: 06 8D A0 3C 83 48 B1 45   6C DC 5A 04 9E CF 0C A1  ...<.H.El.Z.....
0010: 56 D2 B4 34                                        V..4
]
]

#2: ObjectId: 2.5.29.19 Criticality=false
BasicConstraints:[
  CA:true
  PathLen:2147483647
]

#3: ObjectId: 2.5.29.15 Criticality=false
KeyUsage [
  Key_CertSign
  Crl_Sign
]



*******************************************
*******************************************
```

`kubectl get infinispan mycluster -n infinispan  -o yaml`
```yaml
apiVersion: infinispan.org/v1
kind: Infinispan
metadata:
  annotations:
    infinispan.org/monitoring: "true"
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"infinispan.org/v1","kind":"Infinispan","metadata":{"annotations":{},"name":"mycluster","namespace":"infinispan"},"spec":{"configListener":{"enabled":false},"container":{"cpu":"1000m","extraJvmOpts":"-Dinfinispan.bind.address=0.0.0.0","memory":"1024Mi"},"logging":{"categories":{"org.infinispan":"debug","org.jgroups":"debug"}},"replicas":2,"security":{"endpointAuthentication":true,"endpointEncryption":{"certSecretName":"mycluster-tls-secret","clientCert":"None","type":"Secret"},"endpointSecretName":"infinispan-dg-identities"},"service":{"container":{"storage":"1024Mi","storageClassName":"local-path"},"replicationFactor":2,"sites":{"local":{"encryption":{"protocol":"TLSv1.3","routerKeyStore":{"alias":"mycluster","filename":"keystore.p12","secretName":"mycluster-tls-secret"},"transportKeyStore":{"alias":"mycluster","filename":"keystore.p12","secretName":"mycluster-tls-secret"},"trustStore":{"filename":"keystore.p12","secretName":"mycluster-tls-secret"}},"expose":{"annotations":{"traffic.sidecar.istio.io/excludeInboundPorts":"8888,11223,7900","traffic.sidecar.istio.io/excludeOutboundPorts":"8888,11223,7900"},"nodePort":31314,"type":"NodePort"},"name":"mycluster"},"locations":[{"name":"mycluster","secretName":"infinispan-operator-token-v28c6","url":"openshift://env-vpronkin-rntoweou-cp-01c000a0720319a3.elb.us-east-1.amazonaws.com:443"},{"name":"mycluster-site","secretName":"env-vpronkin-site-xsite-token","url":"openshift://env-vpronkin-site-rqkuvrrv-cp-00350df1736ab712.elb.us-east-1.amazonaws.com:443"}]},"type":"DataGrid"}}}
  creationTimestamp: "2022-07-05T14:01:12Z"
  generation: 2
  name: mycluster
  namespace: infinispan
  resourceVersion: "11949752"
  uid: 4152bb61-4967-45c8-b004-dd5107f5a7a4
spec:
  affinity:
    podAntiAffinity:
      preferredDuringSchedulingIgnoredDuringExecution:
      - podAffinityTerm:
          labelSelector:
            matchLabels:
              app: infinispan-pod
              clusterName: mycluster
              infinispan_cr: mycluster
          topologyKey: r.kubernetes.io/hostname
        weight: 100
  configListener:
    enabled: false
  container:
    cpu: 1000m
    extraJvmOpts: -Dinfinispan.bind.address=0.0.0.0
    memory: 1024Mi
  logging:
    categories:
      org.infinispan: debug
      org.jgroups: debug
  replicas: 2
  security:
    endpointAuthentication: true
    endpointEncryption:
      certSecretName: mycluster-tls-secret
      clientCert: None
      type: Secret
    endpointSecretName: infinispan-dg-identities
  service:
    container:
      storage: 1024Mi
      storageClassName: local-path
    replicationFactor: 2
    sites:
      local:
        encryption:
          protocol: TLSv1.3
          routerKeyStore:
            alias: mycluster
            filename: keystore.p12
            secretName: mycluster-tls-secret
          transportKeyStore:
            alias: mycluster
            filename: keystore.p12
            secretName: mycluster-tls-secret
          trustStore:
            filename: keystore.p12
            secretName: mycluster-tls-secret
        expose:
          annotations:
            traffic.sidecar.istio.io/excludeInboundPorts: 8888,11223,7900
            traffic.sidecar.istio.io/excludeOutboundPorts: 8888,11223,7900
          nodePort: 31314
          type: NodePort
        name: mycluster
      locations:
      - name: mycluster
        secretName: infinispan-operator-token-v28c6
        url: openshift://env-vpronkin-rntoweou-cp-01c000a0720319a3.elb.us-east-1.amazonaws.com:443
      - name: mycluster-site
        secretName: env-vpronkin-site-xsite-token
        url: openshift://env-vpronkin-site-rqkuvrrv-cp-00350df1736ab712.elb.us-east-1.amazonaws.com:443
    type: DataGrid
  upgrades:
    type: Shutdown
status:
  conditions:
  - status: "True"
    type: PreliminaryChecksPassed
  - status: "True"
    type: GossipRouterReady
  - message: 'View: mycluster-0-22504,mycluster-1-45233'
    status: "True"
    type: WellFormed
  - message: Site 'mycluster-site' not ready
    status: "False"
    type: CrossSiteViewFormed
  podStatus:
    ready:
    - mycluster-1
    - mycluster-2
  statefulSetName: mycluster
```  
  
`kubectl get pods -n infinispan`
```sh
NAME                                                      READY   STATUS    RESTARTS   AGE
infinispan-operator-controller-manager-7764c699b6-rk2dg   1/1     Running   0          16m
mycluster-0                                               1/1     Running   0          33m
mycluster-1                                               1/1     Running   0          32m
mycluster-router-67b4bb9fc5-t2bzl                         1/1     Running   0          16m 
```

`kubectl logs -f -n infinispan mycluster-router-67b4bb9fc5-t2bzl`
```sh
GossipRouter started in 2581 ms listening on 0.0.0.0:7900 (TLSv1.3)
Jul 05, 2022 3:54:50 PM org.jgroups.blocks.cs.TcpConnection$Receiver run
WARNING: failed handling message
javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake
        at java.base/sun.security.ssl.SSLSocketImpl.handleEOF(SSLSocketImpl.java:1616)
        at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1434)
        at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1336)
        at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:450)
        at java.base/sun.security.ssl.SSLSocketImpl.ensureNegotiated(SSLSocketImpl.java:841)
        at java.base/sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:932)
        at java.base/java.io.BufferedInputStream.fill(BufferedInputStream.java:252)
        at java.base/java.io.BufferedInputStream.read(BufferedInputStream.java:271)
        at java.base/java.io.DataInputStream.readInt(DataInputStream.java:392)
        at org.jgroups.blocks.cs.TcpConnection$Receiver.run(TcpConnection.java:305)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.io.EOFException: SSL peer shut down incorrectly
        at java.base/sun.security.ssl.SSLSocketInputRecord.read(SSLSocketInputRecord.java:483)
        at java.base/sun.security.ssl.SSLSocketInputRecord.readHeader(SSLSocketInputRecord.java:472)
        at java.base/sun.security.ssl.SSLSocketInputRecord.decode(SSLSocketInputRecord.java:160)
        at java.base/sun.security.ssl.SSLTransport.decode(SSLTransport.java:111)
        at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1426)
        ... 9 more
...

added _mycluster-1-45233 (10.42.0.92:42228) to group xsite
Jul 05, 2022 3:54:58 PM org.jgroups.blocks.cs.TcpConnection$Receiver run
WARNING: failed handling message
javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake
        at java.base/sun.security.ssl.SSLSocketImpl.handleEOF(SSLSocketImpl.java:1616)
        at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1434)
        at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1336)
        at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:450)
        at java.base/sun.security.ssl.SSLSocketImpl.ensureNegotiated(SSLSocketImpl.java:841)
        at java.base/sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:932)
        at java.base/java.io.BufferedInputStream.fill(BufferedInputStream.java:252)
        at java.base/java.io.BufferedInputStream.read(BufferedInputStream.java:271)
        at java.base/java.io.DataInputStream.readInt(DataInputStream.java:392)
        at org.jgroups.blocks.cs.TcpConnection$Receiver.run(TcpConnection.java:305)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.io.EOFException: SSL peer shut down incorrectly
        at java.base/sun.security.ssl.SSLSocketInputRecord.read(SSLSocketInputRecord.java:483)
        at java.base/sun.security.ssl.SSLSocketInputRecord.readHeader(SSLSocketInputRecord.java:472)
        at java.base/sun.security.ssl.SSLSocketInputRecord.decode(SSLSocketInputRecord.java:160)
        at java.base/sun.security.ssl.SSLTransport.decode(SSLTransport.java:111)
        at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1426)
        ... 9 more
...
```

**mycluster-site(env-vpronkin-site) site**
```sh
kubectl get secret -n infinispan mycluster-tls-secret -o jsonpath='{.data.keystore\.p12}' | base64 --decode > keystore.p12
keytool -list -v -storetype pkcs12 -keystore keystore.p12 -storepass password


Keystore type: PKCS12
Keystore provider: SUN

Your keystore contains 1 entry

Alias name: mycluster
Creation date: Jul 5, 2022
Entry type: PrivateKeyEntry
Certificate chain length: 2
Certificate[1]:
Owner:
Issuer: CN=Base - cert-manager CA, O=Digital, ST=Some-State, C=AU
Serial number: 90cecf4d358aec9ce77f876ada6aa5f7
Valid from: Tue Jul 05 16:49:53 EEST 2022 until: Mon Oct 03 16:49:53 EEST 2022
Certificate fingerprints:
         SHA1: A0:EF:70:C5:A9:F3:5C:61:2D:FD:E4:BD:01:6E:AB:8C:01:C1:1C:6E
         SHA256: 78:01:71:FA:07:9B:A7:89:9F:24:56:E1:8C:1B:9E:03:35:3C:37:6B:E2:50:AD:EF:8D:DB:F9:BA:41:37:49:92
Signature algorithm name: SHA256withRSA
Subject Public Key Algorithm: 2048-bit RSA key
Version: 3

Extensions:

#1: ObjectId: 2.5.29.19 Criticality=true
BasicConstraints:[
  CA:false
  PathLen: undefined
]

#2: ObjectId: 2.5.29.15 Criticality=true
KeyUsage [
  DigitalSignature
  Key_Encipherment
]

#3: ObjectId: 2.5.29.17 Criticality=true
SubjectAlternativeName [
  DNSName: mycluster.infinispan.svc
  DNSName: mycluster.infinispan.svc.cluster.local
]

Certificate[2]:
Owner: CN=Base - cert-manager CA, O=Digital, ST=Some-State, C=AU
Issuer: CN=Base - Root CA, O=Digital, ST=Some-State, C=AU
Serial number: 3eac62fdbe71c450323654458d6fce4dec1fc64f
Valid from: Wed Apr 21 09:54:00 EEST 2021 until: Sat Apr 19 09:54:00 EEST 2031
Certificate fingerprints:
         SHA1: C7:95:CF:58:72:4D:AB:3A:E4:12:F6:3C:21:5B:2F:DE:C5:CB:FF:23
         SHA256: 96:3F:EA:78:E8:C5:86:8E:31:7B:54:6A:7A:52:0C:F3:33:13:73:24:D1:18:99:F8:15:24:70:69:B4:ED:EF:2E
Signature algorithm name: SHA256withRSA
Subject Public Key Algorithm: 2048-bit RSA key
Version: 3

Extensions:

#1: ObjectId: 2.5.29.35 Criticality=false
AuthorityKeyIdentifier [
KeyIdentifier [
0000: 06 8D A0 3C 83 48 B1 45   6C DC 5A 04 9E CF 0C A1  ...<.H.El.Z.....
0010: 56 D2 B4 34                                        V..4
]
]

#2: ObjectId: 2.5.29.19 Criticality=false
BasicConstraints:[
  CA:true
  PathLen:2147483647
]

#3: ObjectId: 2.5.29.15 Criticality=false
KeyUsage [
  Key_CertSign
  Crl_Sign
]



*******************************************
*******************************************
```

`kubectl get infinispan mycluster -n infinispan  -o yaml`
```yaml
apiVersion: infinispan.org/v1
kind: Infinispan
metadata:
  annotations:
    infinispan.org/monitoring: "true"
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"infinispan.org/v1","kind":"Infinispan","metadata":{"annotations":{},"name":"mycluster","namespace":"infinispan"},"spec":{"configListener":{"enabled":false},"container":{"cpu":"1000m","extraJvmOpts":"-Dinfinispan.bind.address=0.0.0.0","memory":"1024Mi"},"logging":{"categories":{"org.infinispan":"debug","org.jgroups":"debug"}},"replicas":2,"security":{"endpointAuthentication":true,"endpointEncryption":{"certSecretName":"mycluster-tls-secret","clientCert":"None","type":"Secret"},"endpointSecretName":"infinispan-dg-identities"},"service":{"container":{"storage":"1024Mi","storageClassName":"local-path"},"replicationFactor":2,"sites":{"local":{"encryption":{"protocol":"TLSv1.3","routerKeyStore":{"alias":"mycluster","filename":"keystore.p12","secretName":"mycluster-tls-secret"},"transportKeyStore":{"alias":"mycluster","filename":"keystore.p12","secretName":"mycluster-tls-secret"},"trustStore":{"filename":"keystore.p12","secretName":"mycluster-tls-secret"}},"expose":{"annotations":{"traffic.sidecar.istio.io/excludeInboundPorts":"8888,11223,7900","traffic.sidecar.istio.io/excludeOutboundPorts":"8888,11223,7900"},"nodePort":31314,"type":"NodePort"},"name":"mycluster-site"},"locations":[{"name":"mycluster","secretName":"env-vpronkin-xsite-token","url":"openshift://env-vpronkin-rntoweou-cp-01c000a0720319a3.elb.us-east-1.amazonaws.com:443"},{"name":"mycluster-site","secretName":"infinispan-operator-token-56vc7","url":"openshift://env-vpronkin-site-rqkuvrrv-cp-00350df1736ab712.elb.us-east-1.amazonaws.com:443"}]},"type":"DataGrid"}}}
  creationTimestamp: "2022-07-05T14:01:26Z"
  generation: 2
  name: mycluster
  namespace: infinispan
  resourceVersion: "12001640"
  uid: c6ac7d5c-e464-4ab1-b319-4a4c4d6b64b3
spec:
  affinity:
    podAntiAffinity:
      preferredDuringSchedulingIgnoredDuringExecution:
      - podAffinityTerm:
          labelSelector:
            matchLabels:
              app: infinispan-pod
              clusterName: mycluster
              infinispan_cr: mycluster
          topologyKey: r.kubernetes.io/hostname
        weight: 100
  configListener:
    enabled: false
  container:
    cpu: 1000m
    extraJvmOpts: -Dinfinispan.bind.address=0.0.0.0
    memory: 1024Mi
  logging:
    categories:
      org.infinispan: debug
      org.jgroups: debug
  replicas: 2
  security:
    endpointAuthentication: true
    endpointEncryption:
      certSecretName: mycluster-tls-secret
      clientCert: None
      type: Secret
    endpointSecretName: infinispan-dg-identities
  service:
    container:
      storage: 1024Mi
      storageClassName: local-path
    replicationFactor: 2
    sites:
      local:
        encryption:
          protocol: TLSv1.3
          routerKeyStore:
            alias: mycluster
            filename: keystore.p12
            secretName: mycluster-tls-secret
          transportKeyStore:
            alias: mycluster
            filename: keystore.p12
            secretName: mycluster-tls-secret
          trustStore:
            filename: keystore.p12
            secretName: mycluster-tls-secret
        expose:
          annotations:
            traffic.sidecar.istio.io/excludeInboundPorts: 8888,11223,7900
            traffic.sidecar.istio.io/excludeOutboundPorts: 8888,11223,7900
          nodePort: 31314
          type: NodePort
        name: mycluster-site
      locations:
      - name: mycluster
        secretName: env-vpronkin-xsite-token
        url: openshift://env-vpronkin-rntoweou-cp-01c000a0720319a3.elb.us-east-1.amazonaws.com:443
      - name: mycluster-site
        secretName: infinispan-operator-token-56vc7
        url: openshift://env-vpronkin-site-rqkuvrrv-cp-00350df1736ab712.elb.us-east-1.amazonaws.com:443
    type: DataGrid
  upgrades:
    type: Shutdown
status:
  conditions:
  - status: "True"
    type: PreliminaryChecksPassed
  - status: "True"
    type: GossipRouterReady
  - message: 'View: mycluster-0-15542,mycluster-1-61750'
    status: "True"
    type: WellFormed
  - message: Site 'mycluster' not ready
    status: "False"
    type: CrossSiteViewFormed
  podStatus:
    ready:
    - mycluster-1
    - mycluster-2
  statefulSetName: mycluster
```


`kubectl get pods -n infinispan`
```sh
NAME                                                      READY   STATUS    RESTARTS   AGE
infinispan-operator-controller-manager-7764c699b6-rlbcf   1/1     Running   0          41s
mycluster-0                                               1/1     Running   0          17m
mycluster-1                                               1/1     Running   0          16m
mycluster-router-67b4bb9fc5-mv4mt                         1/1     Running   0          21s
```

`kubectl logs -f -n infinispan mycluster-router-67b4bb9fc5-mv4mt`
```sh
GossipRouter started in 780 ms listening on 0.0.0.0:7900 (TLSv1.3)
Jul 05, 2022 3:55:12 PM org.jgroups.blocks.cs.TcpConnection$Receiver run
WARNING: failed handling message
javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake
        at java.base/sun.security.ssl.SSLSocketImpl.handleEOF(SSLSocketImpl.java:1616)
        at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1434)
        at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1336)
        at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:450)
        at java.base/sun.security.ssl.SSLSocketImpl.ensureNegotiated(SSLSocketImpl.java:841)
        at java.base/sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:932)
        at java.base/java.io.BufferedInputStream.fill(BufferedInputStream.java:252)
        at java.base/java.io.BufferedInputStream.read(BufferedInputStream.java:271)
        at java.base/java.io.DataInputStream.readInt(DataInputStream.java:392)
        at org.jgroups.blocks.cs.TcpConnection$Receiver.run(TcpConnection.java:305)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.io.EOFException: SSL peer shut down incorrectly
        at java.base/sun.security.ssl.SSLSocketInputRecord.read(SSLSocketInputRecord.java:483)
        at java.base/sun.security.ssl.SSLSocketInputRecord.readHeader(SSLSocketInputRecord.java:472)
        at java.base/sun.security.ssl.SSLSocketInputRecord.decode(SSLSocketInputRecord.java:160)
        at java.base/sun.security.ssl.SSLTransport.decode(SSLTransport.java:111)
        at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1426)
        ... 9 more

...

added _mycluster-1-61750 (10.42.4.97:53116) to group xsite
Jul 05, 2022 3:55:26 PM org.jgroups.blocks.cs.TcpConnection$Receiver run
WARNING: failed handling message
javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake
        at java.base/sun.security.ssl.SSLSocketImpl.handleEOF(SSLSocketImpl.java:1616)
        at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1434)
        at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1336)
        at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:450)
        at java.base/sun.security.ssl.SSLSocketImpl.ensureNegotiated(SSLSocketImpl.java:841)
        at java.base/sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:932)
        at java.base/java.io.BufferedInputStream.fill(BufferedInputStream.java:252)
        at java.base/java.io.BufferedInputStream.read(BufferedInputStream.java:271)
        at java.base/java.io.DataInputStream.readInt(DataInputStream.java:392)
        at org.jgroups.blocks.cs.TcpConnection$Receiver.run(TcpConnection.java:305)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.io.EOFException: SSL peer shut down incorrectly
        at java.base/sun.security.ssl.SSLSocketInputRecord.read(SSLSocketInputRecord.java:483)
        at java.base/sun.security.ssl.SSLSocketInputRecord.readHeader(SSLSocketInputRecord.java:472)
        at java.base/sun.security.ssl.SSLSocketInputRecord.decode(SSLSocketInputRecord.java:160)
        at java.base/sun.security.ssl.SSLTransport.decode(SSLTransport.java:111)
        at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1426)
        ... 9 more
...
```

Thank you in advance,
Volodymyr.

Hi @pruivo, 
Yes, you are right, the log message below is related to liveness probe requests ONLY:
```sh
Jul 06, 2022 2:11:48 PM org.jgroups.blocks.cs.TcpConnection$Receiver run
WARNING: failed handling message
javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake
        at java.base/sun.security.ssl.SSLSocketImpl.handleEOF(SSLSocketImpl.java:1616)
        at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1434)
        at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1336)
        at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:450)
        at java.base/sun.security.ssl.SSLSocketImpl.ensureNegotiated(SSLSocketImpl.java:841)
        at java.base/sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:932)
        at java.base/java.io.BufferedInputStream.fill(BufferedInputStream.java:252)
        at java.base/java.io.BufferedInputStream.read(BufferedInputStream.java:271)
        at java.base/java.io.DataInputStream.readInt(DataInputStream.java:392)
        at org.jgroups.blocks.cs.TcpConnection$Receiver.run(TcpConnection.java:305)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.io.EOFException: SSL peer shut down incorrectly
        at java.base/sun.security.ssl.SSLSocketInputRecord.read(SSLSocketInputRecord.java:483)
        at java.base/sun.security.ssl.SSLSocketInputRecord.readHeader(SSLSocketInputRecord.java:472)
        at java.base/sun.security.ssl.SSLSocketInputRecord.decode(SSLSocketInputRecord.java:160)
        at java.base/sun.security.ssl.SSLTransport.decode(SSLTransport.java:111)
        at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1426)
        ... 9 more
```
It would be nice to find a solution to suppress that message because now it is really confusing and litters container logs (probably by updating [/opt/gossiprouter/bin/launch.sh](https://github.com/infinispan/infinispan-images/blob/13.0.x/modules/org/infinispan/runtime/added/gossiprouter/bin/launch.sh) script)



Finally, I was able to setup "xsite TLS encryption" by using keystore and trustore generated by cert-manager (details below), and probably, it would be nice to consider a case when `infinispan-operator` can _re-use_ secret generated by [cert-manager](https://cert-manager.io/next-docs/reference/api-docs/#cert-manager.io/v1.CertificateKeystores) without creating intermediate duplicated secret.

Thanks for your help, 
You can close this issue and create new one to suppress warning message `Caused by: java.io.EOFException: SSL peer shut down incorrectly` inside the GossipRouter POD because of liveness probe requests.

The details of my steps:

```yaml
## Create Certificate resource that generates  'infinispan-dg-keystore' secret with keystore and trustsore inside
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: infinispan-dg-keystore
  namespace: infinispan
spec:
  dnsNames:
  - infinispan-dg
  - infinispan-dg.infinispan.svc
  - infinispan-dg.infinispan.svc.cluster.local
  duration: 2160h0m0s
  issuerRef:
    kind: ClusterIssuer
    name: cert-manager-ca-issuer
  keystores:
    pkcs12:
      create: true
      passwordSecretRef:
        key: password-key
        name: infinispan-dg-keystore-creds
  secretName: infinispan-dg-keystore


## View generated 'infinispan-dg-keystore' secret content
kubectl get secret -n infinispan infinispan-dg-keystore -o yaml

apiVersion: v1
data:
  ca.crt: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURsakNDQW42Z0F3SUJBZ0lVUHF4aS9iNXh4RkF5TmxSRmpXL09UZXdmeGs4d0RRWUpLb1pJaHZjTkFRRUwKQlFBd1lURUxNQWtHQTFVRUJoTUNRVlV4RXpBUkJnTlZCQWdNQ2xOdmJXVXRVM1JoZEdVeEV6QVJCZ05WQkFvTQpDa2RGSUVScFoybDBZV3d4S0RBbUJnTlZCQU1NSDBSRk1pNHdJRVp2ZFc1a1lYUnBiMjRnUW1GelpTQXRJRkp2CmIzUWdRMEV3SGhjTk1qRXdOREl4TURZMU5EQXdXaGNOTXpFd05ERTVNRFkxTkRBd1dqQnBNUXN3Q1FZRFZRUUcKRXdKQlZURVRNQkVHQTFVRUNBd0tVMjl0WlMxVGRHRjBaVEVUTUJFR0ExVUVDZ3dLUjBVZ1JHbG5hWFJoYkRFdwpNQzRHQTFVRUF3d25SRVV5TGpBZ1JtOTFibVJoZEdsdmJpQkNZWE5sSUMwZ1kyVnlkQzF0WVc1aFoyVnlJRU5CCk1JSUJJakFOQmdrcWhraUc5dzBCQVFFRkFBT0NBUThBTUlJQkNnS0NBUUVBejBXemxCaFhCTXZyVG9rZ0VjbHgKcHV3K0dkSDBKZjZOTUM5cmczWWZ5dk5pWE9lVkVMNmlRZGdHVFRSeGltY1lySVJFZytvdzBGL3hFdmIvbDlFRwpQOFNtb3BCUHdoUXVVY0paOVlGWGxuKzR1YkhEbUFJbVduS2s3QUNNQWpWMno3TjlSM2hGV1BtUnRmb0loOWpOCmxlVFh5aHJ0UnJFbjJ1RytSMFl3V01GMnhML0JTdUhtN3o1Sm5nSjkxWTNyd3VqUTFLVk9HeC9VSzNEck52SU4KcTQ3RlhyWGlNRmxiNXVoUkpueTZ6bzFzajdJUzlLMGh5ckxRaUYvTEZJMzdZTEdLRDRnQnFsb1Vlc21qbEFxZAo3SXh2bjJFakpNT28yaU5zenVnT0ZvQTdtK1FWU09sQktLbTBpMlU3NUxDSElzQmo4azhTUFd2Y1lvdVZ3dkVjCjB3SURBUUFCb3o0d1BEQWZCZ05WSFNNRUdEQVdnQlFHamFBOGcwaXhSV3pjV2dTZXp3eWhWdEswTkRBTUJnTlYKSFJNRUJUQURBUUgvTUFzR0ExVWREd1FFQXdJQkJqQU5CZ2txaGtpRzl3MEJBUXNGQUFPQ0FRRUFCY1F3aWhUWApOUGl4NVRpYVNycXorMXdlc2tHcHVyeW9TZkVHdzhnMEEvRVk4ZitSd3lSWjI1R3F2OUV6ckdLRWdaZ3JPZWlUCi9OL2h3a0d1cnAzZ0FOSTRrUmxzdkdGTWFBL1ZoR1hyZnROYUlNWjdRbjVYTU9zOTZjWlBhUTJPeEdKeTdndzkKVDBQMHRxOG1jQU1Wc09sSGNmbEF1TTVNOGtPc09pUCtwNk8vdXk1NXhOQ1JlSUYwdnQ3R2JKQ3Y4b1g0T0M0cgo2Unlmc09KbE1Vc0QwYTEycGYzSmRaNkcrdC82SkMxREtUQ0pkRHpQL3Y4RUdwa2lYSlhXcnRjNXFYOHpkR1lTCnlCbWtMVjFCV3dCNXltT2V2UFNOd2FOS2M3Y1A5WjVCN04wN3pERjFYOURiTGtYYUJVdXpSTjVjZll6Rmo5a2MKVi9oOHNYSGplUHZ4eUE9PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==
  keystore.p12: MIIRSwIBAzCCERcGCSqGSIb3DQEHAaCCEQgEghEEMIIRADCCC7cGCSqGSIb3DQEHBqCCC6gwggukAgEAMIILnQYJKoZIhvcNAQcBMBwGCiqGSIb3DQEMAQYwDgQI1TVfTSUqgoUCAggAgIILcBPlat0+C/NbGfAczNwjLQBtv3xsA7v6bLsPYRYQGjPmgEPQ5TUsx8zS4aiEgEUOc+Mfc3gO4z4A6KDuFeaPsgg/8P461rlb/13uPetLDKuWiAAPVBvjNTJm4nMjz1c0FKn6THxgPy8PuPF1XDsvyyO94PVHpv5BPFkTVdr6ww4tHXfEHcimQ/USQs5VqXTr1tQEkFmvwxnypG7C/0XJxluURdZhX3JgituYbRykYbBcVIvn4LQ1egxvXQlIEPxR+7JuH18fttVbET5S6CBRCBgsrl7e9b2noBeTZbiMSQSGh3fqK0Od1ZMqrvT3mVBGI1qv1anawHChzRL+CGwuvoJzkqSak8/eCfWy1dbE3PKyKtb3UdYiYDbeEKRmGCQ8QbcPLS+iIkO1gUPu0QiSRZAUuC4dP5+/VU71beWvtKvJq4kps5dcr6+shOp29bIYrmboLY9NMUC6lHElUOA9s5jhytSkR88J+N7nErHfxziRx+VQTTDV6YgXYyHZ0pbnyojiiIf/sBEmKYOp4hpTkU3dIor0tLHvUCJXYZ8rx3Boc263A/6FNHFhUEUg+xm1tAze2dKWf5hP58SLIciO1nv3NDy4Zo3YXBg9oFskaCk9on0fMLN3AebYogLjRu/tXVm9SD9ngPwuwpIfN/0umDrKDDxqIm6grqRJb0euGzStRn518bRrhPeXchBn23BL2fmUO4Av5oV1OzAuCWHP+SnqEHPtL6zpI8EIK4gVxsoVhMk5Bkp1kq18XunCYq97eFBsBnF4+r5CqoExn+ebgsvvaAz5bL0Z6b5yqRAMjH2jTlrw5r5TNVfAYUTEL4UBYYK5R//R9hzlzvKEXKMh201Zo9cj5ZVQlA1RRpVq4oXBQcoTn5OGbFwT/EttI+Wm0yjJcjnnX9oWyOnIUsMWnvoNt08rOJV+tXpgGn9pKRU8YCHSkxdf8ujJ7eoZXfuxYIlpujKX/Mmn9QZUShK9TuJGZZDmCJ+gPxpVK5D9U7zN88PLzuDRW8N6z/8BiKFWrPCdqkFFmxHX5x7hvgYKDj9M24bH54TX8C4J4nU3E5K1mNhgCKLv91IARsALsa8bCviHqJl67AlcOB5JVV/s/PMecM3zUIpx6r5Dx360dIzs1A+3O3chNMMBGvuokWW1T3xYLy+X2ceRswrO1U70BluCbD8ulRIah6M8TizfJwupOLrByKjZ31l0INNEu7ixfstiw/IiCBUfEohbw5I1xbUzRUI6IXgCKcpPO5cHmCxhRy5tpf7/ogrU4JMymLEhej0h9KvCji1tFTrSxehCQ+Fg2g6/zApRGJoDN2UyVxmPrCxkLB9Vb4T15R7zTebOFx0V7dDXBoO5a/jtA3xL5HKxFLZ3dhJns6+QCdgO8mAdej3nCk4fnQD+gtXxceg9v98bwqDtj9Aj3khYa6xuQgvo21qDZ3kF9pSIhs+zvHIijjY0ZVngr0xxGtXf0hJlVcOFbPDSZztuI1FFEKftjP7ka2/159tP1oYh3a7IPLDIVi10RV8omtQR/LoP184M4HN5WGXR95Fn5XqoEZphtB3ah0tC4D7PVXfgc83h+TVzhyjSsvVfm1sbxH0cVjqBFZDxhG9/k6NgHYXJ2Z2zLnxI4KUC8SJjgShIBRqUujT7lKJuylkJQajXH1Svch2UpH7/5YS13sVXm1T6FbWyLeL+z1BrXJ/3GOKbHaEoYK/WCOCD731yShKIwSZ2UjAzu2XA4dV/Vdnx6rE88u7SVcFXHQC6icvvpew0/BjUTT9/yOWpi7JaMLQE2+HGxMNDMXG666KgyQxtmDxUQB0XsgutWXlilQq3iKFHPA78CKRj5BChb03LCYWoDEUlQvy3rlJ53t+DCVjpMm4jX+r66SOHb68+ZN0sjpkGh8H9cbPOETowRkig4ddn41DCvjwvY2ZMf+S5cqy+l6RVYihchhkgOoaCTiCLX61zmu9Xb+UO/gcEk0FfXpY8OC2Fj3FoWL3CrR78MfSjFyWV8nYC4cjMFBY42JeQfFrd70t91Okl+/O2JCt1YxnUZWG35GdU+mY856RRSKTgycIWfcb2CHzCd1QLGkn9OQ4M/ZgkMJgFfsEfw87JzWBEEQLs0bfUCVgTVfHoNGQunip2lLlyxdEE1tQUdUy5VATg9VaEQRSRVZ2t1RwlrCHmxrUqMgFVn/oyKhRLNnInbmp3TzjhnN3bdJ5Cxm7rhlDNGtULZOPTSA4ZPjk/cUpNTuJmBoQnhn+L49G0Rb/J1evhbjQbKqGqITTLYI4UmeU/xqNu7VNiJqB6oY+ABs8J8CteO0CuW+Nze7yPTo024vr+v28aknmY0DdNj6k+ZYXKs78Zm8Qy+weatoaBjhbLpF6zO7SenTN7XMRKUewUH1Lqtv+h+Pr9HLOaGw91sw1ko06iQnpalA2Ds+7CZWVFgDT7+wHSRQKgNxJlFY6ojw8oer/LhXnp9HklE2wYkA4BvBpJl/ogVTz5lswQV9tq9mNRs6y62TM9VDur5Zfypon3IbNQBQnfBqy/4lxpoq1stXWp7toAcHmTR/DstQD2teT//ebcH3Bn2Y8KtfW7Z8rT3dXmS+zlHok6vWinsn/b9lvbDVUuejTSXR9ncrVvtmh1xTUXQBm0v6uLkBmV9NRSRYaOcqGrENCtRfJerDwgr/jruz8CvKfVjjWJbO+YHk3SFPnzdph+0Z/1RedocsehmmbpEKe9uZbbyHb+AB0NtAtcl0chBZPpRtygEvfImQ37aPjgfKRSrORRyt8Hq+Ojnb7RlLfvGfwkRdHSIRgEXVk+OJ1XE7sjuAgPjiaypOrSCPQak480ToP+HLjkHpD4UnqTtwg4s5sl1sY49XTZ1FAqoSzlEmQjRHdfRIFiADI+JowYJkeQVeFmTP0YM9RX2IkYaJgF+jgTQPto1SPUoDO0j3HRVpuxt5BH6gf8qworutBvkEBjZS1vhuKBFt6ibHmf7HfNOdC4bC13ulHs5sFHpY5EzRORUe2WBr7qGs50zE3d5VvypIpmA5ghSS/HYpHMxquL3YeQpT8gbkBRg543oCV4MQcdXO42OTfrq+0KoklMujBe2MlF74NknOpyzTo8yI4SzYJPB4csTISfrUuV1QLEDYoJlFi/d/qA5fEppXoJr+d8dB8efVQQPSsliIXv+e2Gap661CKuVpFBj0a7nWDRMqNcoEohpmDCuXog++eUb/yXuIdycLLBmTmTjdjGBCWyDxPO2AQh7cNbjzwo4w2PIVAiesk95RTenXC0J03bpTg1kLC/Z0XAu1OhhI3fTNBKV721gbN3MtbEH+SE9IuljvT+aTT08kReIpJ1uox4l8CefKAptiW6x3fnNLbh4FJ49iHTcRN+fYaeXCDPTWy1XYYLHC6yg5t5NXebUmjIjk4g/83RnNxXZ483lBHkxLb0N6ny7nGj3ATkaN1POPAEOEvowjzV+gpnF65V3I5107TFH6pgyC0qe5pNQcy4+DS+AXz5iB8W8wqNOzxncMmr+aUrUTjV/6tB/emGDMwTJLibWugOa+d7krEbXdOOM+5T6IRIyKRFFcUeijFlXAYXGPUDjoI7fsKiUQS6mAXkUG+ygFnLIBqgVcxJDhHHU2qezmcI5/PnSE7J5jUzdV80msselpn7nffSXXQucqz8RYvBuZiFtHgeo1+Tbkmo6ZMtp/Jyd868S18L7WBZbuSuGZCTGQrv7dEYgi8snwDeItp4iX0WWorylfD/8ztodZ+WjcQfetfwYY09vWSG32AFeLvDdBiusMCIdO5QE2+1Zn6lfrTSP/JrRre8TerVtUF3G4EaqhmKKEVXYgdSJhoVH+3fhBYIxgl1t75UM/mm5XPNbqLkJvHMb4+j8sE7wAdO7bENhTiinxETwJPK1lFdtPTU4sRIB95VzvtJODfAzTCCBUEGCSqGSIb3DQEHAaCCBTIEggUuMIIFKjCCBSYGCyqGSIb3DQEMCgECoIIE7jCCBOowHAYKKoZIhvcNAQwBAzAOBAiqh1qYeqgKTwICCAAEggTIZ8BFtSl0KJvXYUUddHfPxmryzjCTjynbMMd0q/fGuHakBnYV8lq93dAwoPuvx0Tr9CAE7OulknJq7Kk868R3kc9C3MweRqepIBlguZTb4v20o/zmYgSb0Btc6IMSqrng4nU+IZFj4R7liPyf3NWs7eZbkgU/ylonCvEb9ML6PtzQ4ScjpwftrhS/Xy0DPWyVYcquGktRRHajQ5uil0LSwyvAMDmqnE9rpJ3CsHVfv6A9kORdq8pRbWeDmVCrcjEnSnBgRm7Ofh6HiDtWGU5KibgMI53VWBxtsMufbr8EVdrIgZju2CuCoIckvyDvlnITDmPw2WPTkE5ixc8JHrCPq3lJkJA5ORoeXrJxYAv1Jlm1TakuwAZ24vFBoeFmMywKY6iMLfVlZbEDyOOxOQu68GEch+sM6lxERdA0rwegKcP4m5Jr2T+qr9tR1jaH5dGOrHaGDGzdYMt6AMw4KfdiM0P2PndVDZAmaezYwMMdc7w1LYpnkWMM6tVq/NTVJLx2HtkEdfBZ9js9emAPXZA5f7EStoetK9Z9WnV1ZOVoIH3kRreuYbHfXvOYN5VSpM2JmD96dbI2wDaxac/OwWNMePATWgXigXmzCiU1ksNCZwXAsrc/BuWgTSFXWmXWzmBhWcl58sBqd/pHFgZK+cmNNFnvHLM1ZH5F/ZGa/X5lN7bZzlviJt8Ps1iZixzVKOkZg3GIzsiulJ+aEKeRgu9MGpITjaq3ScqEDuzY66NFRw4hzYvlLPz0UNSuAO5HSuRf9hHoiRglKXU1lP18JtaglT4NECb+Z3Btau+W2D09gmyxfB/n8K69cSXAb3p1pgJNgH007LAQn1VJf6Y5d/X3FXerkCn2oWAq24TV9SUD6RzgYysocyFm6eN92zX2HjBsqb9qn1gnF3KBgVy/C3LWG6qbE49AoLepvTRFgD/XlBM80civO8p8E8YX3CbBZsVpnjulXqt3rN7tTnHDWiehgnVkxpmtl+kKLAJStS3DOtffpBRHHDWuVJvjEsbSSfeX1lq7VnAhj8cD1/0patqmo2iWVqko5LJzdqlsdw0fy+CfqgTeoSX1MPG/avHBEwPcIH/ykA7pMXBO8nxCkuncvPrbLs5Mjy1Q2apudE7HR1xu6bkXDYkhceWos2fTw3dBXqeXYfulSfoJ/2/bh70Z6R+OE+pLg9inFDnP0GP3grifilnAqi/D2MOkmoxoZqqugr+O/XzkW6LPD5903aFOHelu4kBvsffxBTcEqDFYHTLjiEpK19tkj/pOknLZECrK9BdzPPG6CeYW4maZEyrGJOhmd2ACwz8Ke0pZHyHB0JyFj9/JhSR+BEwziioSjEgh0aKz5q+VgLM3v1QRn+InSpajtOWdX5stJKdkDVKbOsa7qmSw+3EeT4vX59/RFTqm4wSmnauQF3xunB+oX09DOSVdwOuVqKq6HoN/1+R7PqMaZFhmWYUD/r1fZrDrHponXHdj8BFFdxNJXddKU2QXZ5PPG7PdD20loReWHkEHG1RKXzYKtfFe95EN/tmnsJw+P8IsSLT/QERmg8KDMcyNG3nvd6T1t9FlHSv3Z0Su9R9mWX8TKDPHlaW3TVO0ofM7h/3VtE2qLZGmZbX69fsQigXZhIrRyhV1MSUwIwYJKoZIhvcNAQkVMRYEFFvO6+rFxexfXrERjC0T9jpgOXSlMCswHzAHBgUrDgMCGgQUc6YyUBdE0No+IN0KH8+UB/n/o4YECMZH5OAkXBjx
  tls.crt: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURmVENDQW1XZ0F3SUJBZ0lSQVBiUCtUbXROWStvOGVjaGdmcTlLOG93RFFZSktvWklodmNOQVFFTEJRQXcKYVRFTE1Ba0dBMVVFQmhNQ1FWVXhFekFSQmdOVkJBZ01DbE52YldVdFUzUmhkR1V4RXpBUkJnTlZCQW9NQ2tkRgpJRVJwWjJsMFlXd3hNREF1QmdOVkJBTU1KMFJGTWk0d0lFWnZkVzVrWVhScGIyNGdRbUZ6WlNBdElHTmxjblF0CmJXRnVZV2RsY2lCRFFUQWVGdzB5TWpBM01EWXhNVE15TWpWYUZ3MHlNakV3TURReE1UTXlNalZhTUFBd2dnRWkKTUEwR0NTcUdTSWIzRFFFQkFRVUFBNElCRHdBd2dnRUtBb0lCQVFDa3lxU09mWkRaT2dhS2lreGQ2UFZSMTRjZQpwU2xFOUpiU3FCRSt6SVMwanliYWg5dkw5MjIyM3dPVUdVTkpnUTAxV05HUlNrWkoxRjU5MUR3U1RhWmpSem54CnlxSFlkanlVMmFYaU5zbGJzblJMMkhONlNEcXdodSsyVk5MdWZQRk12dWovYUo1NlBpUkI0LzhoM3V0V0J5Y0kKTFVGN01ZQ25aU0pxR2xtWjAwUms2elM4clVhbTFSUWExRlZwRmFMTVMvUC84R25ZTXZCa3NFZHFjWFBOMGhlWApxamVjQldiN2ZpeFZRYkczU2VSNmZRVzB0aDlWd0E2Mk9OZGpRRHVwWWhObUJMZlVsN2FYUUkzWStrNWYvaTVPClVGRVA5UzhWdHB4UEprVDEzVktqWmlmNER3WlVhY0NVQlk3TEt5R1FISDZ4b0NSaDZOWGdXUXJ5WXEzL0FnTUIKQUFHamdZZ3dnWVV3RGdZRFZSMFBBUUgvQkFRREFnV2dNQXdHQTFVZEV3RUIvd1FDTUFBd1pRWURWUjBSQVFILwpCRnN3V1lJTmFXNW1hVzVwYzNCaGJpMWtaNEljYVc1bWFXNXBjM0JoYmkxa1p5NXBibVpwYm1semNHRnVMbk4yClk0SXFhVzVtYVc1cGMzQmhiaTFrWnk1cGJtWnBibWx6Y0dGdUxuTjJZeTVqYkhWemRHVnlMbXh2WTJGc01BMEcKQ1NxR1NJYjNEUUVCQ3dVQUE0SUJBUUI4SVYxSkIvbTAzU2kzNW9pZGRLMGp6TzFDaHFoK3R6cWJ3WFBBY2tRKwpqZFBsb1VmNFIraVcyQ2t6REF2OVBJR1Zqa2tha2N1Ky93WWMxUkpqSVVXVnR6YVVzcFNCRzBuM0JrNXh1enh1CjdtVGpMK05vOU9ZNzQ4c2F2Y2pKUy9XajNDMW16NlRFOTZaNmRqWHR1TXUrTmt3YXZIYlBhaElaVmNHM3NpR3gKTzhVVHlqUjI4NHo0S0xQL3dqZFphazRxZzBkR0dCdHNVVHZOUWhadmZsY0RkZjlSQVFrV0lXM05QaDF4cDJHTgo0QVkyVDVDY1c3bm9oUWRENWJuUy9HUTNiWHRxM04wZFNQZGRST21vNE56bGNxME85NEYxZC9hRDlFdFVUNnNGClhIS3BZZ0lRZXM3aUU1ZmtvbENkQmhJOFg5MDh0bFpoaTlrZXNjUVRBcWttCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0KLS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURsakNDQW42Z0F3SUJBZ0lVUHF4aS9iNXh4RkF5TmxSRmpXL09UZXdmeGs4d0RRWUpLb1pJaHZjTkFRRUwKQlFBd1lURUxNQWtHQTFVRUJoTUNRVlV4RXpBUkJnTlZCQWdNQ2xOdmJXVXRVM1JoZEdVeEV6QVJCZ05WQkFvTQpDa2RGSUVScFoybDBZV3d4S0RBbUJnTlZCQU1NSDBSRk1pNHdJRVp2ZFc1a1lYUnBiMjRnUW1GelpTQXRJRkp2CmIzUWdRMEV3SGhjTk1qRXdOREl4TURZMU5EQXdXaGNOTXpFd05ERTVNRFkxTkRBd1dqQnBNUXN3Q1FZRFZRUUcKRXdKQlZURVRNQkVHQTFVRUNBd0tVMjl0WlMxVGRHRjBaVEVUTUJFR0ExVUVDZ3dLUjBVZ1JHbG5hWFJoYkRFdwpNQzRHQTFVRUF3d25SRVV5TGpBZ1JtOTFibVJoZEdsdmJpQkNZWE5sSUMwZ1kyVnlkQzF0WVc1aFoyVnlJRU5CCk1JSUJJakFOQmdrcWhraUc5dzBCQVFFRkFBT0NBUThBTUlJQkNnS0NBUUVBejBXemxCaFhCTXZyVG9rZ0VjbHgKcHV3K0dkSDBKZjZOTUM5cmczWWZ5dk5pWE9lVkVMNmlRZGdHVFRSeGltY1lySVJFZytvdzBGL3hFdmIvbDlFRwpQOFNtb3BCUHdoUXVVY0paOVlGWGxuKzR1YkhEbUFJbVduS2s3QUNNQWpWMno3TjlSM2hGV1BtUnRmb0loOWpOCmxlVFh5aHJ0UnJFbjJ1RytSMFl3V01GMnhML0JTdUhtN3o1Sm5nSjkxWTNyd3VqUTFLVk9HeC9VSzNEck52SU4KcTQ3RlhyWGlNRmxiNXVoUkpueTZ6bzFzajdJUzlLMGh5ckxRaUYvTEZJMzdZTEdLRDRnQnFsb1Vlc21qbEFxZAo3SXh2bjJFakpNT28yaU5zenVnT0ZvQTdtK1FWU09sQktLbTBpMlU3NUxDSElzQmo4azhTUFd2Y1lvdVZ3dkVjCjB3SURBUUFCb3o0d1BEQWZCZ05WSFNNRUdEQVdnQlFHamFBOGcwaXhSV3pjV2dTZXp3eWhWdEswTkRBTUJnTlYKSFJNRUJUQURBUUgvTUFzR0ExVWREd1FFQXdJQkJqQU5CZ2txaGtpRzl3MEJBUXNGQUFPQ0FRRUFCY1F3aWhUWApOUGl4NVRpYVNycXorMXdlc2tHcHVyeW9TZkVHdzhnMEEvRVk4ZitSd3lSWjI1R3F2OUV6ckdLRWdaZ3JPZWlUCi9OL2h3a0d1cnAzZ0FOSTRrUmxzdkdGTWFBL1ZoR1hyZnROYUlNWjdRbjVYTU9zOTZjWlBhUTJPeEdKeTdndzkKVDBQMHRxOG1jQU1Wc09sSGNmbEF1TTVNOGtPc09pUCtwNk8vdXk1NXhOQ1JlSUYwdnQ3R2JKQ3Y4b1g0T0M0cgo2Unlmc09KbE1Vc0QwYTEycGYzSmRaNkcrdC82SkMxREtUQ0pkRHpQL3Y4RUdwa2lYSlhXcnRjNXFYOHpkR1lTCnlCbWtMVjFCV3dCNXltT2V2UFNOd2FOS2M3Y1A5WjVCN04wN3pERjFYOURiTGtYYUJVdXpSTjVjZll6Rmo5a2MKVi9oOHNYSGplUHZ4eUE9PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==
  tls.key: LS0tLS1CRUdJTiBSU0EgUFJJVkFURSBLRVktLS0tLQpNSUlFb2dJQkFBS0NBUUVBcE1xa2puMlEyVG9HaW9wTVhlajFVZGVISHFVcFJQU1cwcWdSUHN5RXRJOG0yb2ZiCnkvZHR0dDhEbEJsRFNZRU5OVmpSa1VwR1NkUmVmZFE4RWsybVkwYzU4Y3FoMkhZOGxObWw0amJKVzdKMFM5aHoKZWtnNnNJYnZ0bFRTN256eFRMN28vMmllZWo0a1FlUC9JZDdyVmdjbkNDMUJlekdBcDJVaWFocFptZE5FWk9zMAp2SzFHcHRVVUd0UlZhUldpekV2ei8vQnAyREx3WkxCSGFuRnp6ZElYbDZvM25BVm0rMzRzVlVHeHQwbmtlbjBGCnRMWWZWY0FPdGpqWFkwQTdxV0lUWmdTMzFKZTJsMENOMlBwT1gvNHVUbEJSRC9VdkZiYWNUeVpFOWQxU28yWW4KK0E4R1ZHbkFsQVdPeXlzaGtCeCtzYUFrWWVqVjRGa0s4bUt0L3dJREFRQUJBb0lCQUFZRzkxL1N6ZUowRWRzdApWYisxam90d3c1dG1WUTBWbVJLWlhSQWI4TnFkeGNFR2FRNmJ2U0tNSTVoZUdDenZFR21jRndZbVhrM0MrN1hFCmRuSUE3dzRBZytrSkRVUmFocnEyL3k4MzY4TnowbmNYc3BtTlhNQXFIMGhlNzl3VEs4d3Z5MXZVR3IvRDJCV2sKM2ljQi9UK0hRWEUySG9mLzlvR28zRXBmTGxuRzVsbVdoNWF2ajFJOUM3T3dETkJsQjBzeHRGT21tZ09NaUJuZgorZ2tzVC9YTkZ2V0FhNUlJOWZramNuYmFIemx3THBqd2JFVWtmMUpwb3B0MUZZY2hWVnpvWXhtclZyc3JTT2pZCmJ3NnBiYXJ4V2dnWDIwcnRoTmFwK2JFVTJIZVhpclhIZjNQVnVBblFENkZzRnhmbFIwN0lUZ2tGSTlMc2ZDbmEKSWZJVEZla0NnWUVBMFFJaDZRVVJldmhIdnc1RHNwOW5ZUTJSektTc1RUS1JTdHN5N1FNVGZqWWxZMHFXbWpkVwpqM1VBR2orMXI0c3A0MGN6VTNlM0pZdW4yM2MxZDNxdjU2b3pnSlpFMlpOc2hRT0d5NnBqNE9jSjNWWDhGRVZXCi81R1hsZzVLcUVSdFZCQjRqSTBEMm1GMTFZNXNQYTNYNWpUdkdmU1BKY2dPOHIxOG1GNmZ5Wk1DZ1lFQXlkZUkKSVBsaHNoSkRwWCtDS2c1ZXVWWmlNNW5BZG82QkhDWU01dS9TQ1hINEJpZFY3aFhxSFhnSGtKeEVIN0k0L2Z1Lwo1dFNSL1c0azZFTTJHemFLNzNqYW9MQ0VWblVTYUZvRVVMLy9ZSUtRZGZ0d0xZckZnMVd1MktURDgxNDlPMnB0CnZoZTRvUEg3SWhSQ3BaaFJ0d1kwck9hWHl3SjdkM0lUUWZZRG5XVUNnWUFFdnRlbVFzY3kwNUJOZGpWelNNdEEKR043R2xyYVN5aU1pWC9lNENlblhZOE5WamN2OWxHUVN2Y1ZTMHo5OHpOd1duUGM5WGVwNm44RWY2b1Q4UVh6Ywo5YkJWcXhQRkhPUEo0VnFlSlRBL1d2K09WdlZ5UUdESlRMVDBZejZ1czE4SUdaZWpHVkNORk1sb0grTzFRczAwCkhOK1E1MHlZdndJWDROV0RZQ0tnb1FLQmdIVEpZTnF6Y1BPMzNheEV1dXZhcExSWVp6bTBITWg4YS9mOWpEbVQKU2RSdFBUVlNUVVlWS0FFbEVqdDZMM0NuanBKbXVBSzN2RUVIR2hQSnorU3ZDblRGV0xyaFo3RWV6Q2ZVN2VwTApMVmlLTTB3UTBEZUpzLzdwOU9FbGRMdThLM0hWZ3VvYWo1bVFhZDZ1R21IMllGR21KWlJsV0M3K29MSGNMNGhhCjhIeUJBb0dBSEZoNERNOVpsYjNFcC9HUzZSSGNVOFE4OGRyMFBIYU1KeGlJWERZWXFDVTVacFlJWEJZTnVwWmEKRTJ1cTlBSVlUZXU1eTJ4RGZYRUNZUjJPRW9nWDlLRnRRbVlhL2xaSkhHYjViUW1JMnkzbVpVU3VGQzhNbVJjOAp3S2RQVFFWdjhZQzRBUituV0M1QlBKbVFqUzVFTDNDckVpOVpTTDZNa1V1L1hQdlA5MEk9Ci0tLS0tRU5EIFJTQSBQUklWQVRFIEtFWS0tLS0tCg==
  truststore.p12: MIIFJgIBAzCCBPIGCSqGSIb3DQEHAaCCBOMEggTfMIIE2zCCBNcGCSqGSIb3DQEHBqCCBMgwggTEAgEAMIIEvQYJKoZIhvcNAQcBMBwGCiqGSIb3DQEMAQYwDgQIE3xWtU9R2doCAggAgIIEkMLUbzoQnfbwqILhK7torJqZyZUBd5Ct322vrefQ8ncrEFD7grQ/ojkb3T+Cq7vCJshBvk3QvrAY70XVsWm/M6m3ZAsyeQaTQMwrTiHoqVuN3rFUMHlkmVjlvjiKt17LH9nkQrUq4hC+H+Lj61hNVLh8BnEl0GKH4AV0Y4F5BQdKjXnPlu5KpqTRz7/wEsLJuVDp2cE8kPMEz1ZpMJfbWqnCD0/tRlqXx/7m+1/SdI6swFPjBjO60EBkHQQKbFdT1l0tliIxU3jul3LzqjZQQlPeCLWfVHDMqs2U3c0sCbqC39wK+3U14tN3Ik3GB+0fk+a4L/gNnurDj1BdiyzI1i2c8wRqO1KP21fqKNO3IqNJc6+NH9wAw8/l0dVU/unIALOJbVeRoQYhU22J7IKJgGNybeI+w0Igp5kjE1YurMOXGZFojvMkuAALLdBMjCZRTJb746dzh00CsSXUiR85aFwXM7TZzYSmdU3yU1natkprdXvsUhOp6ZDu0E+nLBlUZ0RBkFziHN75OTy9tBZWwTsWrzMel4HQ5q5hkgVKShPYQ6HY8YnN40ruztWOsBXdecjf5CAyS5Ata5HnQXl1ZodnMPQa7IFnt+mzXvy1/JyQQXVZsMqkZ2TGL7KMX+gpGoCK7RrTEqqIfoFUSXuXLSB0tDizUU60+95iwH5jvtjdrLzp/ya+UKWl3IN3V3b0/brkuTjyTkCOfzdXWNAjCCQYEjW1+iWVWu9WejQFO0cBuoKROmxe4HS1q7CKJZuT3Vp4xI1y+/+a2OkO38lRuJAb0boeqK/sYB8l2GWiJgzjB6GCWkIleT7quQZu/sZA3qt4Vb0p8lQ84kLMfPBHUKQGOOqeJDD2MkUZG7gaEx93mdKwB1UNabitFrbk2gvQTVL7x//kqJc7e0mQXyfZfzD9AVvTRhrsZ36qMSKGIwIF1NMksy7KmJz56hUD/0s+nNveNmxKFmHIfij5/P5nCBNhcdbwACcGNE05DL2ZvD4FFR6NrTtT6OeUyEyjHojFCCjuPtTU5L05WsO6bRJFmoBy9xjBqe32irBJSgLXhboBmBvpTj/IgPjnUuOOP+O9aD5nuahG8jKaU8mynxe3ULl+9W2snXXEA6/Wqr8juDFFT0YiF7iEO7vu3+Ns08BhkkSAlX4fAHqT9gmaVE16zr0lqnZUJTLrdDenZUZUTf+jbcAseKNj2GZ2PDW5nNWkZY7NY/Ihq2dmCwPL91EObmfLVOxWOKzXKzfWBfWiM+45Qw2TwEcRVlFtVQqAwFa924ZLKFhJ72MFbLAZlMwZ6uuLeExOTmtZ9PZaNAs4rwQkIJAG2Lb9ABHYEODJ4eItR78s3HZ0E622Z+1BLNw3zqcX1+/rclDV8lDGipHsmThHrDzZSt1/FFgZiQELnyCb5O8kszxDI7Jx89QtloNq5Jqc6Eas4rpFcUTB1mdBhKSvfaVEoCOyk7Vgfd92Dbg9/Bl+QaQiKZgxDJvv1kWsIYEqkj9N3Y7lho1q5ztNrtUi8dli5xzLqYHeBgVyOHYh2jlAFyirtCPM4b/Q+F9/XTgwKzAfMAcGBSsOAwIaBBQXOX5MywJ6m8FXUyOUiskDGynLYAQIb64QqbJyYXU=
kind: Secret
metadata:
  name: infinispan-dg-keystore
  namespace: infinispan
type: kubernetes.io/tls


## Create duplicated  'infinispan-dg-keystore-template' secret with format required by infinispan-operator
kubectl get secret -n infinispan infinispan-dg-keystore-template -o yaml
apiVersion: v1
data:
  keystore.p12: MIIRSwIBAzCCERcGCSqGSIb3DQEHAaCCEQgEghEEMIIRADCCC7cGCSqGSIb3DQEHBqCCC6gwggukAgEAMIILnQYJKoZIhvcNAQcBMBwGCiqGSIb3DQEMAQYwDgQI1TVfTSUqgoUCAggAgIILcBPlat0+C/NbGfAczNwjLQBtv3xsA7v6bLsPYRYQGjPmgEPQ5TUsx8zS4aiEgEUOc+Mfc3gO4z4A6KDuFeaPsgg/8P461rlb/13uPetLDKuWiAAPVBvjNTJm4nMjz1c0FKn6THxgPy8PuPF1XDsvyyO94PVHpv5BPFkTVdr6ww4tHXfEHcimQ/USQs5VqXTr1tQEkFmvwxnypG7C/0XJxluURdZhX3JgituYbRykYbBcVIvn4LQ1egxvXQlIEPxR+7JuH18fttVbET5S6CBRCBgsrl7e9b2noBeTZbiMSQSGh3fqK0Od1ZMqrvT3mVBGI1qv1anawHChzRL+CGwuvoJzkqSak8/eCfWy1dbE3PKyKtb3UdYiYDbeEKRmGCQ8QbcPLS+iIkO1gUPu0QiSRZAUuC4dP5+/VU71beWvtKvJq4kps5dcr6+shOp29bIYrmboLY9NMUC6lHElUOA9s5jhytSkR88J+N7nErHfxziRx+VQTTDV6YgXYyHZ0pbnyojiiIf/sBEmKYOp4hpTkU3dIor0tLHvUCJXYZ8rx3Boc263A/6FNHFhUEUg+xm1tAze2dKWf5hP58SLIciO1nv3NDy4Zo3YXBg9oFskaCk9on0fMLN3AebYogLjRu/tXVm9SD9ngPwuwpIfN/0umDrKDDxqIm6grqRJb0euGzStRn518bRrhPeXchBn23BL2fmUO4Av5oV1OzAuCWHP+SnqEHPtL6zpI8EIK4gVxsoVhMk5Bkp1kq18XunCYq97eFBsBnF4+r5CqoExn+ebgsvvaAz5bL0Z6b5yqRAMjH2jTlrw5r5TNVfAYUTEL4UBYYK5R//R9hzlzvKEXKMh201Zo9cj5ZVQlA1RRpVq4oXBQcoTn5OGbFwT/EttI+Wm0yjJcjnnX9oWyOnIUsMWnvoNt08rOJV+tXpgGn9pKRU8YCHSkxdf8ujJ7eoZXfuxYIlpujKX/Mmn9QZUShK9TuJGZZDmCJ+gPxpVK5D9U7zN88PLzuDRW8N6z/8BiKFWrPCdqkFFmxHX5x7hvgYKDj9M24bH54TX8C4J4nU3E5K1mNhgCKLv91IARsALsa8bCviHqJl67AlcOB5JVV/s/PMecM3zUIpx6r5Dx360dIzs1A+3O3chNMMBGvuokWW1T3xYLy+X2ceRswrO1U70BluCbD8ulRIah6M8TizfJwupOLrByKjZ31l0INNEu7ixfstiw/IiCBUfEohbw5I1xbUzRUI6IXgCKcpPO5cHmCxhRy5tpf7/ogrU4JMymLEhej0h9KvCji1tFTrSxehCQ+Fg2g6/zApRGJoDN2UyVxmPrCxkLB9Vb4T15R7zTebOFx0V7dDXBoO5a/jtA3xL5HKxFLZ3dhJns6+QCdgO8mAdej3nCk4fnQD+gtXxceg9v98bwqDtj9Aj3khYa6xuQgvo21qDZ3kF9pSIhs+zvHIijjY0ZVngr0xxGtXf0hJlVcOFbPDSZztuI1FFEKftjP7ka2/159tP1oYh3a7IPLDIVi10RV8omtQR/LoP184M4HN5WGXR95Fn5XqoEZphtB3ah0tC4D7PVXfgc83h+TVzhyjSsvVfm1sbxH0cVjqBFZDxhG9/k6NgHYXJ2Z2zLnxI4KUC8SJjgShIBRqUujT7lKJuylkJQajXH1Svch2UpH7/5YS13sVXm1T6FbWyLeL+z1BrXJ/3GOKbHaEoYK/WCOCD731yShKIwSZ2UjAzu2XA4dV/Vdnx6rE88u7SVcFXHQC6icvvpew0/BjUTT9/yOWpi7JaMLQE2+HGxMNDMXG666KgyQxtmDxUQB0XsgutWXlilQq3iKFHPA78CKRj5BChb03LCYWoDEUlQvy3rlJ53t+DCVjpMm4jX+r66SOHb68+ZN0sjpkGh8H9cbPOETowRkig4ddn41DCvjwvY2ZMf+S5cqy+l6RVYihchhkgOoaCTiCLX61zmu9Xb+UO/gcEk0FfXpY8OC2Fj3FoWL3CrR78MfSjFyWV8nYC4cjMFBY42JeQfFrd70t91Okl+/O2JCt1YxnUZWG35GdU+mY856RRSKTgycIWfcb2CHzCd1QLGkn9OQ4M/ZgkMJgFfsEfw87JzWBEEQLs0bfUCVgTVfHoNGQunip2lLlyxdEE1tQUdUy5VATg9VaEQRSRVZ2t1RwlrCHmxrUqMgFVn/oyKhRLNnInbmp3TzjhnN3bdJ5Cxm7rhlDNGtULZOPTSA4ZPjk/cUpNTuJmBoQnhn+L49G0Rb/J1evhbjQbKqGqITTLYI4UmeU/xqNu7VNiJqB6oY+ABs8J8CteO0CuW+Nze7yPTo024vr+v28aknmY0DdNj6k+ZYXKs78Zm8Qy+weatoaBjhbLpF6zO7SenTN7XMRKUewUH1Lqtv+h+Pr9HLOaGw91sw1ko06iQnpalA2Ds+7CZWVFgDT7+wHSRQKgNxJlFY6ojw8oer/LhXnp9HklE2wYkA4BvBpJl/ogVTz5lswQV9tq9mNRs6y62TM9VDur5Zfypon3IbNQBQnfBqy/4lxpoq1stXWp7toAcHmTR/DstQD2teT//ebcH3Bn2Y8KtfW7Z8rT3dXmS+zlHok6vWinsn/b9lvbDVUuejTSXR9ncrVvtmh1xTUXQBm0v6uLkBmV9NRSRYaOcqGrENCtRfJerDwgr/jruz8CvKfVjjWJbO+YHk3SFPnzdph+0Z/1RedocsehmmbpEKe9uZbbyHb+AB0NtAtcl0chBZPpRtygEvfImQ37aPjgfKRSrORRyt8Hq+Ojnb7RlLfvGfwkRdHSIRgEXVk+OJ1XE7sjuAgPjiaypOrSCPQak480ToP+HLjkHpD4UnqTtwg4s5sl1sY49XTZ1FAqoSzlEmQjRHdfRIFiADI+JowYJkeQVeFmTP0YM9RX2IkYaJgF+jgTQPto1SPUoDO0j3HRVpuxt5BH6gf8qworutBvkEBjZS1vhuKBFt6ibHmf7HfNOdC4bC13ulHs5sFHpY5EzRORUe2WBr7qGs50zE3d5VvypIpmA5ghSS/HYpHMxquL3YeQpT8gbkBRg543oCV4MQcdXO42OTfrq+0KoklMujBe2MlF74NknOpyzTo8yI4SzYJPB4csTISfrUuV1QLEDYoJlFi/d/qA5fEppXoJr+d8dB8efVQQPSsliIXv+e2Gap661CKuVpFBj0a7nWDRMqNcoEohpmDCuXog++eUb/yXuIdycLLBmTmTjdjGBCWyDxPO2AQh7cNbjzwo4w2PIVAiesk95RTenXC0J03bpTg1kLC/Z0XAu1OhhI3fTNBKV721gbN3MtbEH+SE9IuljvT+aTT08kReIpJ1uox4l8CefKAptiW6x3fnNLbh4FJ49iHTcRN+fYaeXCDPTWy1XYYLHC6yg5t5NXebUmjIjk4g/83RnNxXZ483lBHkxLb0N6ny7nGj3ATkaN1POPAEOEvowjzV+gpnF65V3I5107TFH6pgyC0qe5pNQcy4+DS+AXz5iB8W8wqNOzxncMmr+aUrUTjV/6tB/emGDMwTJLibWugOa+d7krEbXdOOM+5T6IRIyKRFFcUeijFlXAYXGPUDjoI7fsKiUQS6mAXkUG+ygFnLIBqgVcxJDhHHU2qezmcI5/PnSE7J5jUzdV80msselpn7nffSXXQucqz8RYvBuZiFtHgeo1+Tbkmo6ZMtp/Jyd868S18L7WBZbuSuGZCTGQrv7dEYgi8snwDeItp4iX0WWorylfD/8ztodZ+WjcQfetfwYY09vWSG32AFeLvDdBiusMCIdO5QE2+1Zn6lfrTSP/JrRre8TerVtUF3G4EaqhmKKEVXYgdSJhoVH+3fhBYIxgl1t75UM/mm5XPNbqLkJvHMb4+j8sE7wAdO7bENhTiinxETwJPK1lFdtPTU4sRIB95VzvtJODfAzTCCBUEGCSqGSIb3DQEHAaCCBTIEggUuMIIFKjCCBSYGCyqGSIb3DQEMCgECoIIE7jCCBOowHAYKKoZIhvcNAQwBAzAOBAiqh1qYeqgKTwICCAAEggTIZ8BFtSl0KJvXYUUddHfPxmryzjCTjynbMMd0q/fGuHakBnYV8lq93dAwoPuvx0Tr9CAE7OulknJq7Kk868R3kc9C3MweRqepIBlguZTb4v20o/zmYgSb0Btc6IMSqrng4nU+IZFj4R7liPyf3NWs7eZbkgU/ylonCvEb9ML6PtzQ4ScjpwftrhS/Xy0DPWyVYcquGktRRHajQ5uil0LSwyvAMDmqnE9rpJ3CsHVfv6A9kORdq8pRbWeDmVCrcjEnSnBgRm7Ofh6HiDtWGU5KibgMI53VWBxtsMufbr8EVdrIgZju2CuCoIckvyDvlnITDmPw2WPTkE5ixc8JHrCPq3lJkJA5ORoeXrJxYAv1Jlm1TakuwAZ24vFBoeFmMywKY6iMLfVlZbEDyOOxOQu68GEch+sM6lxERdA0rwegKcP4m5Jr2T+qr9tR1jaH5dGOrHaGDGzdYMt6AMw4KfdiM0P2PndVDZAmaezYwMMdc7w1LYpnkWMM6tVq/NTVJLx2HtkEdfBZ9js9emAPXZA5f7EStoetK9Z9WnV1ZOVoIH3kRreuYbHfXvOYN5VSpM2JmD96dbI2wDaxac/OwWNMePATWgXigXmzCiU1ksNCZwXAsrc/BuWgTSFXWmXWzmBhWcl58sBqd/pHFgZK+cmNNFnvHLM1ZH5F/ZGa/X5lN7bZzlviJt8Ps1iZixzVKOkZg3GIzsiulJ+aEKeRgu9MGpITjaq3ScqEDuzY66NFRw4hzYvlLPz0UNSuAO5HSuRf9hHoiRglKXU1lP18JtaglT4NECb+Z3Btau+W2D09gmyxfB/n8K69cSXAb3p1pgJNgH007LAQn1VJf6Y5d/X3FXerkCn2oWAq24TV9SUD6RzgYysocyFm6eN92zX2HjBsqb9qn1gnF3KBgVy/C3LWG6qbE49AoLepvTRFgD/XlBM80civO8p8E8YX3CbBZsVpnjulXqt3rN7tTnHDWiehgnVkxpmtl+kKLAJStS3DOtffpBRHHDWuVJvjEsbSSfeX1lq7VnAhj8cD1/0patqmo2iWVqko5LJzdqlsdw0fy+CfqgTeoSX1MPG/avHBEwPcIH/ykA7pMXBO8nxCkuncvPrbLs5Mjy1Q2apudE7HR1xu6bkXDYkhceWos2fTw3dBXqeXYfulSfoJ/2/bh70Z6R+OE+pLg9inFDnP0GP3grifilnAqi/D2MOkmoxoZqqugr+O/XzkW6LPD5903aFOHelu4kBvsffxBTcEqDFYHTLjiEpK19tkj/pOknLZECrK9BdzPPG6CeYW4maZEyrGJOhmd2ACwz8Ke0pZHyHB0JyFj9/JhSR+BEwziioSjEgh0aKz5q+VgLM3v1QRn+InSpajtOWdX5stJKdkDVKbOsa7qmSw+3EeT4vX59/RFTqm4wSmnauQF3xunB+oX09DOSVdwOuVqKq6HoN/1+R7PqMaZFhmWYUD/r1fZrDrHponXHdj8BFFdxNJXddKU2QXZ5PPG7PdD20loReWHkEHG1RKXzYKtfFe95EN/tmnsJw+P8IsSLT/QERmg8KDMcyNG3nvd6T1t9FlHSv3Z0Su9R9mWX8TKDPHlaW3TVO0ofM7h/3VtE2qLZGmZbX69fsQigXZhIrRyhV1MSUwIwYJKoZIhvcNAQkVMRYEFFvO6+rFxexfXrERjC0T9jpgOXSlMCswHzAHBgUrDgMCGgQUc6YyUBdE0No+IN0KH8+UB/n/o4YECMZH5OAkXBjx
  password: bXkta2V5c3RvcmUtcGFzc3dvcmQ=
  truststore.p12: MIIFJgIBAzCCBPIGCSqGSIb3DQEHAaCCBOMEggTfMIIE2zCCBNcGCSqGSIb3DQEHBqCCBMgwggTEAgEAMIIEvQYJKoZIhvcNAQcBMBwGCiqGSIb3DQEMAQYwDgQIE3xWtU9R2doCAggAgIIEkMLUbzoQnfbwqILhK7torJqZyZUBd5Ct322vrefQ8ncrEFD7grQ/ojkb3T+Cq7vCJshBvk3QvrAY70XVsWm/M6m3ZAsyeQaTQMwrTiHoqVuN3rFUMHlkmVjlvjiKt17LH9nkQrUq4hC+H+Lj61hNVLh8BnEl0GKH4AV0Y4F5BQdKjXnPlu5KpqTRz7/wEsLJuVDp2cE8kPMEz1ZpMJfbWqnCD0/tRlqXx/7m+1/SdI6swFPjBjO60EBkHQQKbFdT1l0tliIxU3jul3LzqjZQQlPeCLWfVHDMqs2U3c0sCbqC39wK+3U14tN3Ik3GB+0fk+a4L/gNnurDj1BdiyzI1i2c8wRqO1KP21fqKNO3IqNJc6+NH9wAw8/l0dVU/unIALOJbVeRoQYhU22J7IKJgGNybeI+w0Igp5kjE1YurMOXGZFojvMkuAALLdBMjCZRTJb746dzh00CsSXUiR85aFwXM7TZzYSmdU3yU1natkprdXvsUhOp6ZDu0E+nLBlUZ0RBkFziHN75OTy9tBZWwTsWrzMel4HQ5q5hkgVKShPYQ6HY8YnN40ruztWOsBXdecjf5CAyS5Ata5HnQXl1ZodnMPQa7IFnt+mzXvy1/JyQQXVZsMqkZ2TGL7KMX+gpGoCK7RrTEqqIfoFUSXuXLSB0tDizUU60+95iwH5jvtjdrLzp/ya+UKWl3IN3V3b0/brkuTjyTkCOfzdXWNAjCCQYEjW1+iWVWu9WejQFO0cBuoKROmxe4HS1q7CKJZuT3Vp4xI1y+/+a2OkO38lRuJAb0boeqK/sYB8l2GWiJgzjB6GCWkIleT7quQZu/sZA3qt4Vb0p8lQ84kLMfPBHUKQGOOqeJDD2MkUZG7gaEx93mdKwB1UNabitFrbk2gvQTVL7x//kqJc7e0mQXyfZfzD9AVvTRhrsZ36qMSKGIwIF1NMksy7KmJz56hUD/0s+nNveNmxKFmHIfij5/P5nCBNhcdbwACcGNE05DL2ZvD4FFR6NrTtT6OeUyEyjHojFCCjuPtTU5L05WsO6bRJFmoBy9xjBqe32irBJSgLXhboBmBvpTj/IgPjnUuOOP+O9aD5nuahG8jKaU8mynxe3ULl+9W2snXXEA6/Wqr8juDFFT0YiF7iEO7vu3+Ns08BhkkSAlX4fAHqT9gmaVE16zr0lqnZUJTLrdDenZUZUTf+jbcAseKNj2GZ2PDW5nNWkZY7NY/Ihq2dmCwPL91EObmfLVOxWOKzXKzfWBfWiM+45Qw2TwEcRVlFtVQqAwFa924ZLKFhJ72MFbLAZlMwZ6uuLeExOTmtZ9PZaNAs4rwQkIJAG2Lb9ABHYEODJ4eItR78s3HZ0E622Z+1BLNw3zqcX1+/rclDV8lDGipHsmThHrDzZSt1/FFgZiQELnyCb5O8kszxDI7Jx89QtloNq5Jqc6Eas4rpFcUTB1mdBhKSvfaVEoCOyk7Vgfd92Dbg9/Bl+QaQiKZgxDJvv1kWsIYEqkj9N3Y7lho1q5ztNrtUi8dli5xzLqYHeBgVyOHYh2jlAFyirtCPM4b/Q+F9/XTgwKzAfMAcGBSsOAwIaBBQXOX5MywJ6m8FXUyOUiskDGynLYAQIb64QqbJyYXU=
kind: Secret
metadata:
  creationTimestamp: "2022-07-06T11:37:07Z"
  name: infinispan-dg-keystore-template
  namespace: infinispan
  resourceVersion: "12803544"
  uid: 0bba7851-6750-4cdc-9295-b1034721ba45
type: Opaque


## Add "xsite TLS encryption" section into Infinispan resource
...
        local:
          encryption:
            protocol: TLSv1.3
            routerKeyStore:
              alias: "1"
              filename: keystore.p12
              secretName: infinispan-dg-keystore-template
            transportKeyStore:
              alias: "1"
              filename: keystore.p12
              secretName: infinispan-dg-keystore-template
            trustStore:
              filename: truststore.p12
              secretName: infinispan-dg-keystore-template
...
```

--- a/controllers/constants/constants.go
+++ b/controllers/constants/constants.go
@@ -55,6 +55,7 @@ const (
 	InfinispanUserPort                      = 11222
 	CrossSitePort                           = 7900
 	CrossSitePortName                       = "xsite"
+	GossipRouterDiagPort                    = 7500
 	StatefulSetPodLabel                     = "app.kubernetes.io/created-by"
 	StaticCrossSiteUriSchema                = "infinispan+xsite"
 	CacheServiceFixedMemoryXmxMb            = 200

--- a/pkg/reconcile/pipeline/infinispan/handler/provision/gossiprouter.go
+++ b/pkg/reconcile/pipeline/infinispan/handler/provision/gossiprouter.go
@@ -51,6 +51,8 @@ func GossipRouter(i *ispnv1.Infinispan, ctx pipeline.Context) {
 	}
 	mutateFn := func() error {
 		routerLabels := i.GossipRouterPodLabels()
+		enableJgrpDiag := consts.JGroupsDiagnosticsFlag == "TRUE"
+		var probePort int32 = consts.CrossSitePort
 
 		// if the user configures 0 replicas, shutdown the gossip router pod too.
 		var replicas *int32
@@ -67,6 +69,10 @@ func GossipRouter(i *ispnv1.Infinispan, ctx pipeline.Context) {
 
 		var addKeystoreVolume, addTruststoreVolume bool
 		if i.IsSiteTLSEnabled() {
+			// enable diagnostic for k8s probes
+			enableJgrpDiag = true
+			probePort = consts.GossipRouterDiagPort
+
 			ks := ctx.ConfigFiles().XSite.GossipRouter.Keystore
 			args = append(args, []string{
 				"-nio", "false", // NIO does not work with TLS
@@ -91,6 +97,31 @@ func GossipRouter(i *ispnv1.Infinispan, ctx pipeline.Context) {
 			log.Info("No TLS configured")
 		}
 
+		containerPorts := []corev1.ContainerPort{
+			{
+				ContainerPort: consts.CrossSitePort,
+				Name:          "tunnel",
+				Protocol:      corev1.ProtocolTCP,
+			},
+		}
+
+		if enableJgrpDiag {
+			args = append(args, []string{
+				"-diag_enabled", "true",
+				"-diag_enable_udp", "false",
+				"-diag_enable_tcp", "true",
+				"-diag_port", strconv.Itoa(consts.GossipRouterDiagPort),
+				"-diag_port_range", "0",
+			}...)
+			containerPorts = append(containerPorts,
+				corev1.ContainerPort{
+					ContainerPort: consts.GossipRouterDiagPort,
+					Name:          "tunnel-diag",
+					Protocol:      corev1.ProtocolTCP,
+				},
+			)
+		}
+
 		router.Labels = routerLabels
 		router.Spec = appsv1.DeploymentSpec{
 			Selector: &metav1.LabelSelector{
@@ -104,21 +135,15 @@ func GossipRouter(i *ispnv1.Infinispan, ctx pipeline.Context) {
 				},
 				Spec: corev1.PodSpec{
 					Containers: []corev1.Container{{
-						Name:    GossipRouterContainer,
-						Image:   i.ImageName(),
-						Command: []string{"/opt/gossiprouter/bin/launch.sh"},
-						Args:    args,
-						Env:     []corev1.EnvVar{{Name: "ROUTER_JAVA_OPTIONS", Value: i.Spec.Container.RouterExtraJvmOpts}},
-						Ports: []corev1.ContainerPort{
-							{
-								ContainerPort: consts.CrossSitePort,
-								Name:          "tunnel",
-								Protocol:      corev1.ProtocolTCP,
-							},
-						},
-						LivenessProbe:  TcpProbe(consts.CrossSitePort, 5, 5, 0, 1, 60),
-						ReadinessProbe: TcpProbe(consts.CrossSitePort, 5, 5, 0, 1, 60),
-						StartupProbe:   TcpProbe(consts.CrossSitePort, 5, 1, 1, 1, 60),
+						Name:           GossipRouterContainer,
+						Image:          i.ImageName(),
+						Command:        []string{"/opt/gossiprouter/bin/launch.sh"},
+						Args:           args,
+						Env:            []corev1.EnvVar{{Name: "ROUTER_JAVA_OPTIONS", Value: i.Spec.Container.RouterExtraJvmOpts}},
+						Ports:          containerPorts,
+						LivenessProbe:  TcpProbe(probePort, 5, 5, 0, 1, 60),
+						ReadinessProbe: TcpProbe(probePort, 5, 5, 0, 1, 60),
+						StartupProbe:   TcpProbe(probePort, 5, 1, 1, 1, 60),
 					}},
 				},
 			},

